src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -382,17 +430,145 @@ class ContentExplorer extends Component {
+    };
+
+    disableAllOtherItemButtons = itemToKeepId => {
+        const { handleSelectAllTree, items } = this.props;
+
+        if (this.itemsAreLoading()) {
+            return;
+        }
+
+        items.forEach(item => {
+            item.id === itemToKeepId ? (item.isDisabled = false) : (item.isDisabled = true);
+        });
+        this.setState({ allItemsEnabled: false });
+        handleSelectAllTree(itemToKeepId);
+    };
+
+    enableAllItemButtons = () => {
+        const { handleSelectAllTree, items } = this.props;
+        items.forEach(item => {
+            item.isDisabled = false;
+        });
+        this.setState({ allItemsEnabled: true });
+        handleSelectAllTree(null);
+    };
+
+    handleIncludeSubfoldersToggle = () => {
+        const { allItemsEnabled, includeSubfolders, selectedItems } = this.state;
+        const { handleSelectAllTree } = this.props;
+
+        this.setState({ includeSubfolders: !includeSubfolders }, () => {
+            if (this.shouldDisableAllOtherButtons()) {
+                const selectedItem = Object.values(selectedItems)[0];
+                this.setState({ folderToDeepScan: selectedItem }, () => {
+                    if (this.shouldSelectAndActionDisableItems()) {
+                        this.setState({ itemsAreSelectedAndActionDisabled: true });
+                        handleSelectAllTree(selectedItem.id);
+                    } else {
+                        this.disableAllOtherItemButtons(selectedItem.id);
+                    }
+                });
+            } else if (!allItemsEnabled) {
+                this.setState({ folderToDeepScan: {}, itemsAreSelectedAndActionDisabled: false });
+                this.enableAllItemButtons();
+            }
+        });
+    };
+
+    areOtherItemsDisabled = enabledItemId => {
+        const { items } = this.props;
+
+        const result = items.every(item => {
+            if (
+                (item.id !== enabledItemId && !item.isDisabled) ||
+                (item.id === enabledItemId && item.isDisabled) ||
+                item.isLoading
+            ) {
+                return false;
+            }
+            return true;
+        });
+        return result;
+    };
+
+    foldersPresent = () => {
+        const { items } = this.props;
+        for (let i = 0; i < items.length; i += 1) {
+            if (items[i].type === TYPE_FOLDER) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    numOfSelectedItems = () => {
+        const { selectedItems } = this.state;
+        return Object.keys(selectedItems).length;
+    };
+
+    noFoldersSelected = () => {
+        const { selectedItems } = this.state;
+        const result = Object.values(selectedItems).every(item => {
+            if (item.type === TYPE_FOLDER) {
+                return false;
+            }
+            return true;
+        });
+        return result;
+    };
```

Comment:
You can use `isFolderPresent` here.

Reply:
Ah yeah if I make isFoldersPresent take in an iterable then I can use it for both. The difference here is that I am checking if any of the selected items (state.selectedItems) are folders where as foldersPresent is currently checking if any of the displayed items (props.items) are folders